### PR TITLE
Convert to lib

### LINKED ISSUE #13: Change to library and add u4pak as bin
As I understand it this project can be changed to a library and the current main + cli support files be placed in ```src/bin/u4pak``` to have a library that also builds an executable.

### PR DESCRIPTION
**Pull request by Xenira**
Converted project to lib with binary.
Closes #13 

Requires #11 to be merged first.

[Changes from #11](https://github.com/panzi/rust-u4pak/pull/14/commits/c295a25fb398dfcf0e13e4fc3caf2a7e21315850)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,10 +1,18 @@
 [package]
 name = "u4pak"
 version = "1.3.0"
-authors = ["Mathias Panzenböck <grosser.meister.morti@gmx.net>"]
+authors = ["Mathias Panzenböck <grosser.meister.morti@gmx.net>", "L. Sprengel <l.sprengel@pm.me>"]
```
Comment (Xenira) [inline Cargo.toml:4]: Hope its ok, that I added myself here :)

Comment (panzi) [reply Cargo.toml:4]: Of course!

**Review by panzi — APPROVED** [review]

**Comment by panzi** [comment]
Oh, did you never compile under Linux? Because it doesn't even compile anymore.

```
error[E0583]: file not found for module `mount`
  --> src/bin/u4pak/main.rs:49:1
   |
49 | pub mod mount;
   | ^^^^^^^^^^^^^^
   |
   = help: to create the module `mount`, create file "src/bin/u4pak/mount.rs" or "src/bin/u4pak/mount/mod.rs"

error[E0432]: unresolved imports `mount::mount`, `mount::MountOptions`
  --> src/bin/u4pak/main.rs:51:17
   |
51 | pub use mount::{mount, MountOptions};
   |                 ^^^^^  ^^^^^^^^^^^^ no `MountOptions` in `mount`
   |                 |
   |                 no `mount` in `mount`

error[E0063]: missing field `encryption_key` in initializer of `Options`
   --> src/bin/u4pak/main.rs:936:17
    |
936 |                 Options {
    |                 ^^^^^^^ missing `encryption_key`
```

**Comment by Xenira** [comment]
Ohhh, no. Did work under windows, as I need UE for my tests.

**Comment by Xenira** [comment]
Should be an easy fix though.

**Comment by Xenira** [comment]
1. Move `pub mod mount;` and `pub use mount::*` to `src/lib.rs`
2. Now it should be importable using `use u4pak::mount`

**Comment by panzi** [comment]
Done. The fuse stuff is Linux only as I wrote it, since I only have Linux. Dunno if its possible to support other OSes with that feature. It is handy, though. You just mount a huge archive and browse it with your regular file browser to view/extract single files. :smile:

**Pull request merged**